740 implement forgot password feature

keshrinandan99 opened the pull request:
## Description
<!-- Link your ticket using #{ISSUE_NUMBER}. And add a clear and concise description of what this PR does. -->
#740 
This pull request introduces a comprehensive password reset functionality that allows users to securely regain access to their accounts through email verification, eliminating the need for support intervention.


## What type of PR is this? (Check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📄 Documentation Update
- [ ] 👨‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🛠️ CI/CD

## Screenshots (if applicable)
<!-- Add screenshots to help explain your 
[forgetPassword.webm](https://github.com/user-attachments/assets/d521f1a5-7686-4572-b990-446d904ccab0)




## Checklist
- [x] I have performed a self-review of my code  
- [ ] I have commented my code, particularly in hard-to-understand areas  
- [ ] I have added tests that prove my fix is effective or that my feature works  
- [x] New and existing unit tests pass locally with my changes

keshrinandan99 committed:
feat: add reset password functionality and integrate it into the app

- Implemented ResetPassword component for handling password resets.
- Updated App component to include a new route for password reset.
- Enhanced user experience with form validation and error handling during the reset process.
- Added necessary API calls for token validation and password reset actions.

keshrinandan99 committed:
refactor: update ResetPassword component and remove legacy code

- Refactored ResetPassword component for improved structure and readability.
- Removed deprecated resetPassword.js file and adjusted imports accordingly.
- Updated cookie settings in signinController for enhanced security.
- Changed base URL in api_instance.js to use environment variable for better configuration.
- Enhanced form validation logic in ResetPassword component.

keshrinandan99 committed:
Merge branch 'stage' into 740-implement-forgot-password-feature

keshrinandan99 commented:
[forgetPassword.webm](https://github.com/user-attachments/assets/5c8d2244-5388-4875-9b68-90b34ac74b53)

aps08 commented:
@TanishqSingla

aps08 reviewed (changes requested):
Where are the test files ?

keshrinandan99 commented:
@aps08 I am working on it .

keshrinandan99 committed:
test: add unit tests for ResetPassword component

- Created comprehensive tests for the ResetPassword component to ensure proper functionality.
- Included tests for rendering, form validation, successful password reset, and error handling.
- Utilized mocking for API calls and navigation to simulate user interactions and responses.

aps08 commented:
> @aps08 I am working on it .

If are still working on this PR then why raise a PR ?

keshrinandan99 committed:
added resetPassword test

keshrinandan99 commented:
@aps08 Added the ui test for reset Password .
<img width="1165" height="117" alt="image" src="https://github.com/user-attachments/assets/e0172c4b-45c4-45e4-9b9d-2898202bfa27" />

keshrinandan99 committed:
test: enhance ResetPassword component tests with additional scenarios

keshrinandan99 commented:
@aps08 updated the resetPassword test. 
<img width="951" height="98" alt="testssdone" src="https://github.com/user-attachments/assets/11848bbf-9e71-47d2-b458-5927e309d956" />

printgourav committed:
Merge branch 'stage' into 740-implement-forgot-password-feature

TanishqSingla commented on packages/ui/__tests__/components/auth/ResetPassword.test.jsx:
```diff
@@ -0,0 +1,426 @@
+      expect(screen.getByLabelText("New Password")).toBeInTheDocument();
+      expect(screen.getByLabelText("Confirm Password")).toBeInTheDocument();
+      expect(
+        screen.getByRole("button", { name: BUTTON_TEXT.submit })
```
`getByRole` is already a exception that can reject the test

keshrinandan99 replied:
getByRole is preferred for interactive elements therefore I used that against getByText.

TanishqSingla commented on packages/ui/src/components/auth/ResetPassword.jsx:
```diff
@@ -0,0 +1,237 @@
+
+  return (
+    <div
+      style={{
```
Why are we using inline styles here?

TanishqSingla commented on packages/ui/src/components/auth/ResetPassword.jsx:
```diff
@@ -0,0 +1,237 @@
+      }}
+    >
+      <div
+        style={{
```
Same as above

TanishqSingla commented on packages/ui/src/components/auth/ResetPassword.jsx:
```diff
@@ -0,0 +1,237 @@
+              <div className={styles["form-width"]}>
+                <CustomInput
+                  type="password"
+                  name="newPassword"
```
don't see any point of name attribute here, since we're using JS to submit the form. Let me know if I'm missing something

keshrinandan99 replied:
name attribute is passed to handlefocus & handleChange

keshrinandan99 committed:
refactor: simplify ResetPassword component layout and improve styling

keshrinandan99 committed:
refactored inline styles

keshrinandan99 commented:
@TanishqSingla  I have moved the inline styles to custom css classes .

printgourav committed:
Revert "Merge branch 'stage' into 740-implement-forgot-password-feature"

This reverts commit 5217b46b3d7ca00c58ccd77686adedaaff928cde, reversing
changes made to feafbb3e243a8ddf02bc15710fe720e7c6c6ac57.

Idontnol commented on packages/ui/src/components/hero/HeroSection.module.css:
```
- background-image: url("/hero-background.png");
+ background-image: url("hero-background.png");
```
Hey @keshrinandan99 is there any specific reason for this change ? 
The leading slash is important to ensure path is root relative. without it the background image wont render correctly in production. I suggest you to revert this change.

keshrinandan99 replied:
@Idontnol @printgourav committed this change it went unnoticed by me .

printgourav replied:
Thanks @Idontnol  for pointing this out might happened while reverting did'nt noticed that time though thanks for letting me know about this @keshrinandan99, hope this had been fixed. for now

keshrinandan99 committed:
Delete packages/ui/src/components/hero/HeroSection.module.css

Idontnol commented on packages/ui/src/components/hero/HeroSection.module.css:
```diff
@@ -1,95 +0,0 @@
-    height: 200px;
-    width: 200px;
-  }
-}
```
Hey, why did you delete the entire HeroSection.module.css file?

keshrinandan99 committed:
feat: add HeroSection back mistakenly deleted

keshrinandan99 committed:
fix: update label for current password field in change password form

aps08 commented on packages/ui/__tests__/components/auth/ResetPassword.test.jsx:
```diff
@@ -0,0 +1,426 @@
+import { render, screen, fireEvent, waitFor } from "@testing-library/react";
```
Check other files containing test. Follow the same standard.

aps08 commented on packages/ui/src/components/auth/ResetPassword.jsx:
```diff
@@ -0,0 +1,211 @@
+      toast.success("Password reset successful! Please sign in.");
+      navigate("/");
+    } else {
+      setLocalErrorMsg(resetErrorMsg || "Failed to reset password.");
```
are we not getting message from backend that we are hardcoding the message here ?

aps08 committed:
reafctored code

aps08 committed:
Merge branch 'stage' into 740-implement-forgot-password-feature
